modules/quantcastBidAdapter.js

```diff
@@ -1,128 +1,168 @@
-    let domain = loc.hostname;
-    let publisherId = 0;
+    const bidRequestsList = bids.map(bid => {
+      const bidSizes = [];
 
-    publisherId = '' + bids[0].params.publisherId;
-    utils._each(bids, function(bid) {
-      let key = bid.placementCode;
-      var bidSizes = [];
-      utils._each(bid.sizes, function (size) {
+      bid.sizes.forEach(size => {
         bidSizes.push({
-          'width': size[0],
-          'height': size[1]
+          width: size[0],
+          height: size[1]
         });
       });
 
-      bidRequests[key] = bidRequests[key] || {
-        'publisherId': publisherId,
-        'requestId': bid.bidId,
-        'bidId': bid.bidId,
-        'site': {
-          'page': loc.href,
-          'referrer': referrer,
-          'domain': domain,
+      // Request Data Format can be found at https://wiki.corp.qc/display/adinf/QCX
+      const requestData = {
+        publisherId: bid.params.publisherId,
+        requestId: bid.bidId,
+        imp: [
+          {
+            banner: {
+              battr: bid.params.battr,
+              sizes: bidSizes
+            },
+            placementCode: bid.placementCode,
+            bidFloor: bid.params.bidFloor || DEFAULT_BID_FLOOR
+          }
+        ],
+        site: {
+          page: loc.href,
+          referrer,
+          domain
         },
-        'imp': [{
-
-          'banner': {
-            'battr': bid.params.battr,
-            'sizes': bidSizes,
-          },
-          'placementCode': bid.placementCode,
-          'bidFloor': bid.params.bidFloor || DEFAULT_BID_FLOOR,
-        }]
+        bidId: bid.bidId
       };
-    });
 
-    utils._each(bidRequests, function (bidRequest) {
-      ajax.ajax(QUANTCAST_CALLBACK_URL, $$PREBID_GLOBAL$$.handleQuantcastCB, JSON.stringify(bidRequest), {
+      const data = JSON.stringify(requestData);
+
+      const url =
+        bid.params.publisherId === QUANTCAST_TEST_PUBLISHER
+          ? publisherTagURLTest
+          : publisherTagURL;
+
+      return {
+        data,
         method: 'POST',
+        url,
         withCredentials: true
-      });
+      };
     });
-  }
 
-  // Export the `callBids` function, so that Prebid.js can execute
-  // this function when the page asks to send out bid requests.
-  return {
-    callBids: callBids,
-    QUANTCAST_CALLBACK_URL: QUANTCAST_CALLBACK_URL
-  };
-};
+    return bidRequestsList;
+  },
+
+  /**
+   * Function get called when the browser has received the response from Quantcast server.
+   * The function parse the response and create a `bidResponse` object containing one/more bids.
+   * Returns an empty array if no valid bids
+   *
+   * Response Data Format can be found at https://wiki.corp.qc/display/adinf/QCX
+   *
+   * @param {*} serverResponse A successful response from Quantcast server.
+   * @return {Bid[]} An array of bids which were nested inside the server.
+   *
+   */
+  interpretResponse(serverResponse) {
```

Comment:
#1742 got merged recently.

The first argument to interpretResponse now looks like this:

{
  body: responseBody,
  headers: {
    get: function(header) { /* returns a header from the HTTP response */ }
  }
}
You'll have to pull master and update the spec so that it looks digs into that object as well now.

Reply:
Updated with bee2344 and 417852e